### PR TITLE
Fix all integrations imports for v4.

### PR DESCRIPTION
**Pull request by purple4reina**
This pull request aims to switch all imports that were v3 to v4 in the shim.

There were two changes required to get all the tests to pass. I left comments in the code detailing these. Otherwise, all changes are simply `s/v3/v4`.

### CODE REVIEW

File: build-script.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ for dir in $DIRS; do
   if [ $dir == "." ]; then
     rm -rf v3/
     rm -rf v4/
-  elif [[ $dir =~ "v3" ]]; then
+  elif [[ $dir =~ "v3/" ]]; then
```
Comment (purple4reina) [inline build-script.sh:36]: This change is required for the `v4/integrations/nrecho-v3` tests.

The problem here was that the v3 at the end of the package name was matching this if statement unexpectedly. The solution is to throw a slash at the end to ensure that we are talking about a v3 directory and not a v3 module name.

File: v4/integrations/nrgin/go.mod [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
+module github.com/newrelic/go-agent/v4/integrations/nrgin
 
 // As of Dec 2019, the gin go.mod file uses 1.12:
 // https://github.com/gin-gonic/gin/blob/master/go.mod
 go 1.12
 
 require (
-	github.com/gin-gonic/gin v1.4.0
-	github.com/newrelic/go-agent/v3 v3.0.0
+	github.com/gin-gonic/gin v1.5.0
```
Comment (purple4reina) [inline v4/integrations/nrgin/go.mod:8]: When running the nrgin tests with gin v1.4.0 the following error was raised. I saw this error in the github actions logs, when running the tests locally in docker, and when running the tests locally outside of docker.

```
##[error]../../../../../../../pkg/mod/github.com/gin-gonic/gin@v1.4.0/binding/msgpack.go:12:2: ambiguous import: found package github.com/ugorji/go/codec in multiple modules:
	github.com/ugorji/go v1.1.4 (/home/runner/work/go-agent/go-agent/pkg/mod/github.com/ugorji/go@v1.1.4/codec)
	github.com/ugorji/go/codec v0.0.0-20181204163529-d75b2dcb6bc8 (/home/runner/work/go-agent/go-agent/pkg/mod/github.com/ugorji/go/codec@v0.0.0-20181204163529-d75b2dcb6bc8)
```

The error did not occur when running the nrgin tests against the v3 agent.

I was unable to track down the exact reason why this failure was happening. The `github.com/gin-gonic/gin` package requires `github.com/ugorji/go v1.1.4 ` in its go.mod file. I checked everywhere to try to find out which package was requiring the `github.com/ugorji/go/codec v0.0.0-20181204163529-d75b2dcb6bc8` but couldn't find it. Go's tooling for this wasn't very helpful.

After spending about an hour and a half on trying to find the cause, I decided to switch to solutions. For this I found that simply upgrading `github.com/gin-gonic/gin` to v1.5.0 did the trick. v1.4.0 of gin was released on May 7, 2019 and v1.5.0 was released on Nov 27, 2019. The most recent version of gin is v1.6.3. You can see the full [release notes here](https://github.com/gin-gonic/gin/releases).

File: v4/integrations/nrecho-v3/README.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
-# v3/integrations/nrecho-v3 [![GoDoc](https://godoc.org/github.com/newrelic/go-agent/v3/integrations/nrecho-v3?status.svg)](https://godoc.org/github.com/newrelic/go-agent/v3/integrations/nrecho-v3)
+# v4/integrations/nrecho-v3 [![GoDoc](https://godoc.org/github.com/newrelic/go-agent/v4/integrations/nrecho-v3?status.svg)](https://godoc.org/github.com/newrelic/go-agent/v4/integrations/nrecho-v3)
```
Comment (pyohannes) [inline v4/integrations/nrecho-v3/README.md:1]: Should those changes also be made in `integrations/nrecho-v4/README.md`?

Comment (purple4reina) [reply v4/integrations/nrecho-v3/README.md:1]: Great eye!

**Pull request merged**